controllers/vsphereclusteridentity_controller.go

```diff
@@ -119,6 +119,12 @@ func (r clusterIdentityReconciler) Reconcile(ctx _context.Context, req reconcile
 		return ctrl.Result{}, r.reconcileDelete(ctx, identity)
 	}
 
+	// Add a finalizer and requeue to ensure that the secret is deleted when the identity is deleted.
+	if !ctrlutil.ContainsFinalizer(identity, infrav1.VSphereClusterIdentityFinalizer) {
+		ctrlutil.AddFinalizer(identity, infrav1.VSphereClusterIdentityFinalizer)
+		return reconcile.Result{Requeue: true}, nil
```

Comment:
```suggestion
		return reconcile.Result{}, nil
```
The update will requeue automatically (the godoc is maybe a bit misleading in this regard)